cdap-cli/src/main/java/co/cask/cdap/cli/command/StartProgramCommand.java

```diff
@@ -56,23 +56,24 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
       throw new CommandInputError(this);
     }
 
-    String appId = programIdParts[0];
-    String programName = programIdParts[1];
-    ProgramId programId = cliConfig.getCurrentNamespace().app(appId).program(elementType.getProgramType(), programName);
+    ProgramId programId = parseProgramId(arguments, elementType);
+    String appName = programId.getApplication();
+    String appVersion = programId.getVersion();
+    String programName = programId.getProgram();
 
-    String runtimeArgsString = arguments.get(ArgumentName.RUNTIME_ARGS.toString(), "");
+    String runtimeArgsString = arguments.getOptional(ArgumentName.RUNTIME_ARGS.toString(), "");
     if (runtimeArgsString == null || runtimeArgsString.isEmpty()) {
       // run with stored runtime args
-      programClient.start(programId.toId(), isDebug);
-      runtimeArgsString = GSON.toJson(programClient.getRuntimeArgs(programId.toId()));
-      output.printf("Successfully started %s '%s' of application '%s' with stored runtime arguments '%s'\n",
-                    elementType.getName(), programName, appId, runtimeArgsString);
+      programClient.start(programId, isDebug, null);
```

Comment:
This is currently the only method that accepts ProgramId to start a program, may add more methods to start a program in the future.